scanner/image/scanner/download-mappings.sh

```diff
@@ -20,11 +25,25 @@ urls=(
 for url in "${urls[@]}"; do
     filename=$(basename "$url")
     echo "Downloading ${url} > ${output_dir}/$filename"
-    curl --retry 3 -sSL --fail -o "${output_dir}/$filename" "$url"
+    curl --location --silent --fail --show-error --retry 3 --max-time 60 \
+        --output "${output_dir}/$filename" "$url"
     if [[ ! (-s "${output_dir}/$filename") ]]; then
         echo "${output_dir}/$filename is empty"
         exit 1
     fi
+
+    if command -v jq &>/dev/null; then
+        echo "Validating if ${output_dir}/$filename contains parseable JSON"
+        if jq -e . >/dev/null 2>&1 < "${output_dir}/$filename"; then
+            echo "Validated"
+        else
+            echo "${output_dir}/$filename is not valid JSON"
+            exit 1
+        fi
+    else
+        echo "Could not find jq to validate JSON"
+        echo "WARNING: Skipping JSON validation"
+    fi
```

Comment:
I actually originally wrote this with both jq and python in mind, but wanted to err on the side of pragmatism and try to get this in ASAP for the release. I'll push my original implementation, and we can decide collectively if that's the way to go (plus, it'll trigger the konflux build, so we can see those results as well).

Reply:
I don't feel like I'm eager to approve the original implementation.
By my comment I implicitly meant: can you please find a way to make the check more concise? And I offered a way to reduce it by 13 lines. The original code now is a lot longer. It seems to do the same but has more branching and hence needs more effort to verify.

If you're really attached to it and eager to merge it, I would not object in case you secure Scanner team's approval but I do think longer code hurts maintainability.

Reply:
Gotcha. What'd you think about this version? I tried to make it compatible with Python 2 and 3

Reply:
Python 2? It's dead, no need to support it.